unittests/deep_mind_tests.cpp

```diff
@@ -53,363 +50,44 @@ struct deep_mind_tester : deep_mind_log_fixture, validating_tester
-               result = {weight, 0, 1};
-            }
-         }
-         return result;
-      };
-      auto generate_output = [&](std::size_t i, std::size_t j){
-         std::vector<std::string> result;
-         while(i != 0 && j != 0)
-         {
-            switch(tab[i - 1][j - 1].prev)
-            {
-            case 1:
-               --i;
-               result.push_back(escape_line(*(line_iter + i)));
-               break;
-            case 2:
-               --j;
-               break;
-            case 3:
-               --i, --j;
-               result.push_back(merge_line(*(line_iter + i), *(pattern_iter + j)));
-            }
-         }
-         while(i-- > 0)
-         {
-            result.push_back(escape_line(*(line_iter + i)));
-         }
-         std::reverse(result.begin(), result.end());
-         return result;
-      };
-      auto next = [&](std::ptrdiff_t i, std::ptrdiff_t j) {
-         auto output = generate_output(i, j);
-         line_iter += i;
-         pattern_iter += j;
-         all_output.insert(all_output.end(), output.begin(), output.end());
-      };
-      auto editdiff_loop = [&](){
-         // Fill an expanding square with equal lines and patterns
-         for(std::ptrdiff_t i = 0; i < std::min(line_end - line_iter, pattern_end - pattern_iter); ++i)
-         {
-            tab.emplace_back();
-            for(std::ptrdiff_t j = 0, j_end; j < i; ++j)
-            {
-               tab[i].push_back(add_point(i, j));
-               if(tab[i][j].unmodified >= resync_threshold)
-               {
-                  next(i + 1, j + 1);
-                  return;
-               }
-            }
-            for(std::ptrdiff_t j = 0; j < i; ++j)
-            {
-               tab[j].push_back(add_point(j, i));
-               if(tab[j][i].unmodified >= resync_threshold)
-               {
-                  next(j + 1, i + 1);
-                  return;
-               }
-            }
-            tab[i].push_back(add_point(i, i));
-            if(tab[i][i].unmodified >= resync_threshold)
-            {
-               next(i + 1, i + 1);
-               return;
-            }
-         }
-         // fill excess lines
-         for(std::ptrdiff_t i = pattern_end - pattern_iter; i < line_end - line_iter;++i)
-         {
-            tab.emplace_back();
-            for(std::ptrdiff_t j = 0; j < pattern_end - pattern_iter; ++j)
-            {
-               tab[i].push_back(add_point(i, j));
-            }
-         }
-         // fill excess patterns
-         for(std::ptrdiff_t i = line_end - line_iter; i < pattern_end - pattern_iter; ++i)
-         {
-            for(std::ptrdiff_t j = 0; j < line_end - line_iter; ++j)
-            {
-               tab[j].push_back(add_point(j, i));
-            }
-         }
-         next(line_end - line_iter, pattern_end - pattern_iter);
-      };
-      editdiff_loop();
+   while(std::getline(input_file, s)) {
+      output_file << s << "\n";
```

Comment:
Can use `output_file << input_file.rdbuf()` or just `fc::copy` and get rid of the streams entirely.

Reply:
Thanks for the suggestion. It further simplifies the code. I changed to use `fc:rename` as `fc:copy` does not allow destination to exist before copying.